### PR TITLE
issue: 1465058 Introduce ring per ip-address logic

### PR DESCRIPTION
**Pull request by liranoz12**

### CODE REVIEW

**Comment by liranoz12** [comment]
@igor-ivanov please review

**Comment by liranoz12** [comment]
bot:retest

File: src/vma/sock/sockinfo.cpp [hunk 1]
```diff
@@ -596,12 +596,9 @@ net_device_resources_t* sockinfo::create_nd_resources(const ip_address ip_local)
 
 		unlock_rx_q();
 		m_rx_ring_map_lock.lock();
-		resource_allocation_key *key;
-		if (m_rx_ring_map.size()) {
-			key = m_ring_alloc_logic.get_key();
-		} else {
-			key = m_ring_alloc_logic.create_new_key();
-		}
+		resource_allocation_key *key = m_rx_ring_map.size() && m_ring_alloc_logic.is_logic_support_migration() ?
```
Comment (rafiw) [inline src/vma/sock/sockinfo.cpp]: this should not work
please check key value after this line

File: src/vma/sock/sockinfo.cpp [hunk 2]
```diff
@@ -667,7 +664,10 @@ bool sockinfo::destroy_nd_resources(const ip_address ip_local)
 		// Release ring reference
 		BULLSEYE_EXCLUDE_BLOCK_START
 		unlock_rx_q();
-		if (!p_nd_resources->p_ndv->release_ring(m_ring_alloc_logic.get_key())) {
+		resource_allocation_key *key = m_ring_alloc_logic.is_logic_support_migration() ?
```
Comment (rafiw) [inline src/vma/sock/sockinfo.cpp]: same as above

File: src/vma/dev/ring_allocation_logic.cpp [hunk 3]
```diff
@@ -118,6 +121,12 @@ resource_allocation_key* ring_allocation_logic::create_new_key(int suggested_cpu
 		}
 	}
 
+	if (m_res_key.get_ring_alloc_logic() == RING_LOGIC_PER_IP) {
```
[the remark this answers is not in the document]
Comment (igor-ivanov) [reply src/vma/dev/ring_allocation_logic.cpp]: no

**Comment by liranoz12** [comment]
@rafiw

File: src/vma/dev/ring_allocation_logic.h [hunk 4]
```diff
@@ -88,10 +97,10 @@ class ring_allocation_logic_rx : public ring_allocation_logic
 {
 public:
 	ring_allocation_logic_rx():ring_allocation_logic(){}
-	ring_allocation_logic_rx(int fd, resource_allocation_key &ring_profile, const void* owner):
+	ring_allocation_logic_rx(source_t source, resource_allocation_key &ring_profile, const void* owner):
```
Comment (rafiw) [inline src/vma/dev/ring_allocation_logic.h]: add & to source

File: src/vma/dev/ring_allocation_logic.h [hunk 5]
```diff
@@ -100,10 +109,10 @@ class ring_allocation_logic_tx : public ring_allocation_logic
 {
 public:
 	ring_allocation_logic_tx():ring_allocation_logic(){}
-	ring_allocation_logic_tx(int fd, resource_allocation_key &ring_profile, const void* owner):
+	ring_allocation_logic_tx(source_t source, resource_allocation_key &ring_profile, const void* owner):
```
Comment (rafiw) [inline src/vma/dev/ring_allocation_logic.h]: add &

**Comment by liranoz12** [comment]
bot:retest

**Pull request merged**